src/Features/Core/Navigation/IDocumentNavigationService.cs

```diff
@@ -25,16 +25,16 @@ internal interface IDocumentNavigationService : IWorkspaceService
         /// <summary>
         /// Navigates to the given position in the specified document, opening it if necessary.
         /// </summary>
-        bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, bool usePreviewTab = false);
+        bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, bool reopenDocument = false, bool usePreviewTab = false);
```

Comment:
It looks like we already have a helper for getting the provisional tab contextual status. Should we just say we reopen if/when the document is in a provisional tab but the state says not to open one as such?